src/oidc-add/parse.c

```diff
@@ -0,0 +1,37 @@
+#include "parse.h"
+#include "defines/agent_values.h"
+#include "defines/ipc_values.h"
+#include "defines/oidc_values.h"
+#include "utils/json.h"
+#include "utils/key_value.h"
+#include "utils/printer.h"
+#include "utils/stringUtils.h"
+#include "wrapper/list.h"
+
+struct loaded_accounts_response parseForLoadedAccountsListResponse(char* response) {
+  if (response == NULL) {
+    return (struct loaded_accounts_response){NULL};
+  }
+
+  INIT_KEY_VALUE(IPC_KEY_STATUS, OIDC_KEY_ERROR, "info");
+  if (CALL_GETJSONVALUES(response) < 0) {
+    printError("Read malformed data. Please hand in bug report.\n");
+    secFree(response);
+    SEC_FREE_KEY_VALUES();
+    return (struct loaded_accounts_response){NULL};
+  }
+
+  secFree(response);
+  KEY_VALUE_VARS(status, error, info);
+  if (_error) {  // error
+    oidc_errno = OIDC_EERROR;
+    oidc_seterror(_error);
+    SEC_FREE_KEY_VALUES();
+    return (struct loaded_accounts_response){NULL};
+  } else {
+    secFree(_status);
+    oidc_errno        = OIDC_SUCCESS;
+    char *accounts = JSONArrayStringToDelimitedString(_info, "||");
```

Comment:
Any ideas on how to better store the result? 2D array?

Reply:
Wouldn't it be a solution to just return it as a JSON Array string?

Reply:
Yes, it would be a cleaner way to do so. On the other hand, putty (or any other software that will use the library) will need to deal with JSON strings instead of split by delimiter

Reply:
That's right, so for the moment I think this is fine.
We could use another delimiter. Space delimited feels more in line with other delimited strings in the agent.